src/component-importer/importcomponentdialog.cpp

```diff
@@ -0,0 +1,123 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.pl/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "importcomponentdialog.h"
+
+#include <QWidget>
+#include <QDir>
+
+#include <coreplugin/icore.h>
+
+#include <utils/qtcassert.h>
+
+#include <projectexplorer/session.h>
+#include <projectexplorer/project.h>
+#include <projectexplorer/projectnodes.h>
+
+using namespace Asn1Acn::Internal::ComponentImporter;
+
+ImportComponentDialog::ImportComponentDialog(QWidget *parent)
+    : QDialog(parent)
+    , m_libraries(Settings::load<Settings::Libraries>())
+{
+    m_ui.setupUi(this);
+
+    setAttribute(Qt::WA_DeleteOnClose);
+    connect(Core::ICore::instance(), &Core::ICore::coreAboutToClose, this, &QWidget::close);
+
+    connect(m_libraries.get(), &Settings::Libraries::changed, this, &ImportComponentDialog::refreshPaths);
```

Comment:
I feel obliged to mention that this connect is useless now, as changed signal is not emitted anywhere. I did not wanted to spend time on fixing it as this part may, or may not be used later. In final implementation it should be done though.

Reply:
ok, add some `// TODO` here so we'll find it during some clean up :)